apps/babelstream/package.py

```diff
@@ -0,0 +1,268 @@
+    # version("4.0", sha256="a9cd39277fb15d977d468435eb9b894f79f468233f0131509aa540ffda4f5953")
+    version('main', branch='main')
+    maintainers = ["robj0nes ", "tomdeakin", "tom91136"]
+    
+    #Languages
+    variant("sycl", default=False, description="Enable SYCL support")
+    variant("sycl2020", default=False, description="Enable SYCL support")
+    variant("omp", default=False, description="Enable OpenMP support")
+    variant("ocl", default=False, description="Enable OpenCL support")
+    variant("tbb", default=False, description="Enable TBB support")
+    variant("acc", default=False, description="Enable OpenACC support") 
+    variant("thrust", default=False, description="Enable THRUST support")
+    variant("raja", default=False, description="Enable RAJA support") 
+    variant("stddata", default=False, description="Enable STD-data support")
+    variant("stdindices", default=False, description="Enable STD-indices support")
+    variant("stdranges", default=False, description="Enable STD-ranges support")
+    
+    variant("dir", values=str, default="none", description="Enable Directory support")
+    
+    #Kokkos conflict and variant
+    conflicts("dir=none",when="+kokkos",msg="KOKKKOS requires architecture to be specfied by dir=")
+    variant("kokkos", default=False, description="Enable KOKKOS support")
+    
+    #ACC conflict
+    variant("cpu_arch", values=str, default="none", description="Enable CPU Target for ACC")
+    variant("target", values=str, default="none", description="Enable CPU Target for ACC")
+    
+    #STD conflicts
+    conflicts("+stddata",when="%gcc@:10.1.0",msg="STD-data requires newer version of GCC") 
+    conflicts("+stdindices",when="%gcc@:10.1.0",msg="STD-indices requires newer version of GCC")
+    conflicts("+stdranges",when="%gcc@:10.1.0",msg="STD-ranges requires newer version of GCC")
+    
+    # CUDA conflict 
+    conflicts("cuda_arch=none",when="+cuda",msg="CUDA requires architecture to be specfied by cuda_arch=")
+    variant("mem", values=str, default="DEFAULT", description="Enable MEM Target for CUDA")
+        
+    #download raja from https://github.com/LLNL/RAJA
+    conflicts("dir=none",when="+raja",msg="RAJA requires architecture to be specfied by dir= [#download raja from https://github.com/LLNL/RAJA]")
+
+    #Thrust Conflict
+    conflicts("~cuda", when="+thrust", msg="Thrust requires +cuda variant")
+
+    #TBB Dependency
+    depends_on("intel-oneapi-tbb", when="+tbb")
+    
+    #Kokkos Dependency
+    depends_on("kokkos@3.7.1", when="+kokkos")
+
+    #OpenCL Dependency
+    depends_on("pocl@1.5", when="+ocl")
+  
+    # this flag could be used in all required languages
+    variant("flags", values=str, default="none", description="Additional CXX flags to be provided")
+
+    # SYCL variant
+    variant("compiler", values=str, default="none", description="Enable Compiler Target for SYCL")
```

Comment:
I expect this will be pushed back on by the main Spack developers as there is another notation for choosing a compiler.
Can we detect this with the `%icpx` for Intel/`%syclcc`for OpenSYCL (hipSYCL)?

Reply:
I am afraid it is not possible to detect custom values with `%` symbol but is it OK to rename `compiler` to something like `compiler-option` ?